index.bs

```diff
@@ -342,36 +342,55 @@ The <dfn attribute for="Document">onprerenderingchange</dfn> attribute is an [=e
+      * |navigationType| is "`other`"
+      * |request|'s [=request/method=] is \``GET`\`
+      * |browsingContext|'s [=active document=]'s [=Document/prerendering browsing contexts map=][(|request|'s [=request/URL=], |request|'s [=request/referrer policy=])] [=map/exists=] and is not [=prerendering browsing context/empty=]
 
     then:
 
-    1. Let |successorBC| be |browsingContext|'s [=active document=]'s [=Document/prerendering browsing contexts map=][(|resource|'s [=request/URL=], |resource|'s [=request/referrer policy=])].
+    1. Return true.
+
+  1. Otherwise, return false.
+
+</div>
+
+Patch the [=navigate=] algorithm to allow the [=prerendering browsing context/activate|activation=] of a [=prerendering browsing context=] in place of a normal navigation as follows:
+
+<div algorithm="navigate activate patch">
+  In [=navigate=], append the following steps after the fragment navigation handling (currently step 6):
 
-    1. If |successorBC| is not [=prerendering browsing context/empty=], then:
+  1. If |resource| is a [=request=] and [=can activate a prerender=] given |browsingContext|, |historyHandling|, <var ignore>navigationType</var>, and |resource| returns true, then:
 
-      1. [=prerendering browsing context/Activate=] |successorBC| in place of |browsingContext| given |historyHandling|.
+    1. [=prerendering browsing context/Activate=] <var ignore>successorBC</var> in place of |browsingContext| given |historyHandling|.
 
-      1. Return.
+    1. Return.
 </div>
 
+Navigation redirects can also [=prerendering browsing context/activate=] [=prerendering browsing contexts=]. This is defined in the [[#redirect-handling]] section.
+
 <h3 id="redirect-handling">Redirect handling</h3>
 
-Patch the redirect-handling portion of the [=process a navigate fetch=] algorithm to update certain properties of a [=prerendering browsing context=] upon cross-origin redirects as follows:
+This section contains two types of changes to the navigation redirect handling portion of the [=process a navigate fetch=] algorithm:
+ * Logic allowing navigation redirects that are not inside of a [=prerendering browsing context=] to [=prerendering browsing context/activate=] a [=prerendering browsing context=].
+ * Logic that updates specific state of a [=prerendering browsing context=] when it redirects coss-origin.
+
+Patch the [=process a navigate fetch=] algorithm like so:
 
 <div algorithm="navigate activate patch redirect handling">
-  In [=process a navigate fetch=], append the following steps after the first sub-step under "While true:" in order to handle redirects correctly;
+  In [=process a navigate fetch=], append the following steps after the first sub-step under "While true:" in order to handle redirects correctly:
+
+  1. If |browsingContext| is not a [=prerendering browsing context=] and [=can activate a prerender=] given |browsingContext| |historyHandling|, <var ignore>navigationType</var>, and <var ignore>request</var> returns true, then:
```

Comment:
We should either modify this PR to do one of the following, for completeness:
 - Restrict prerendering activation in the non-redirect nav case to non-prerendering browsing contexts (so that prerender requests cannot activate other prerender browsing contexts)
 - Lift this "is not a prerendering browsing context" condition, so that all navigation requests (including prerender browsing context navs) can activate activation prerendering browsing contexts

Reply:
> Restrict prerendering activation in the non-redirect nav case to non-prerendering browsing contexts (so that prerender requests cannot activate other prerender browsing contexts)

Recursive prerendering seems complicated and scary to me, but I can't think of any concrete problems with it. This might be a thing where we disallow it for now, with a spec note that we could loosen it in the future? I guess it largely comes down to our implementation appetite.

Reply:
That sounds good to me. I'm cool with preventing it for now, especially since it isn't really the intention of prerendering, and could potentially be a footgun anyways. OK I'll go for that then.